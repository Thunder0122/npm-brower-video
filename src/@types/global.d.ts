import { SingletonRouter } from "next/router";
import { NextContext, NextComponentType } from "@types/next";
import { MyMobxStore } from "../stores";
import TranslationsStore from "../stores/TranslationsStore";

export interface AppNextRootPageProps {
  t: i18next.TranslationFunction;
  router: SingletonRouter;
  translationLanguageFullCode?: string;
  defaultLanguageFullCode: string;
}

export interface AppWithIdNextRootPageProps<T> extends AppNextRootPageProps {
  data: T;
  query: { id: string };
  translationsStore: TranslationsStore;
}

export interface AppNextRootPageGetInitialProps {
  req?: any;
  translationLanguageFullCode?: string;
  defaultLanguageFullCode: string;
}

export interface AppWithIdNextRootPageGetInitialProps
  extends AppNextRootPageGetInitialProps {
  mobxStore?: MyMobxStore;
  query: { id: string };
}

export interface CustomNextContext<Q extends DefaultQuery = DefaultQuery> {
  mobxStore: MyMobxStore; // SPECIFIC
  translationLanguageFullCode?: string; // SPECIFIC
  defaultLanguageShortCode: string; // SPECIFIC
  defaultLanguageFullCode: string; // SPECIFIC
  /** path section of URL */
  pathname: string;
  /** query string section of URL parsed as an object */
  query: Q;
  /** String of the actual path (including the query) shows in the browser */
  asPath: string;
  /** HTTP request object (server only) */
  req?: http.IncomingMessage;
  /** HTTP response object (server only) */
  res?: http.ServerResponse;
  /** Fetch Response object (client only) - from https://developer.mozilla.org/en-US/docs/Web/API/Response */
  jsonPageRes?: NodeResponse;
  /** Error object if any error is encountered during the rendering */
  err?: Error;
}

export interface CustomNextAppContext<Q extends DefaultQuery = DefaultQuery> {
  Component: NextComponentType<any, any, NextContext<Q>>;
  router: RouterProps<Q>;
  ctx: CustomNextContext<Q>;
}

export class CustomNextApp<P = {}> extends React.Component<
  P & DefaultAppIProps & AppProps
> {
  mobxStore: MyMobxStore;
  static getInitialProps(context: NextAppContext): Promise<DefaultAppIProps>;
}
