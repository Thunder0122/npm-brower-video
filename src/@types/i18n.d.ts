export type LanguageList = { code: string; label: string }[];
