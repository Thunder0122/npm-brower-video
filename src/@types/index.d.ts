export * from "./tmdb";
export * from "./global";
export * from "./i18n";
