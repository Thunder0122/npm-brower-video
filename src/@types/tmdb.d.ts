/**
 * Based on generated types made with https://jvilk.com/MakeTypes/
 */
export interface TmdbMovieEntity {
  adult: boolean;
  backdrop_path: string;
  belongs_to_collection?: null;
  budget: number;
  genres?: (TmdbGenresEntity)[] | null;
  homepage: string;
  id: number;
  imdb_id: string;
  original_language: string;
  original_title: string;
  overview: string;
  popularity: number;
  poster_path: string;
  production_companies?: (TmdbProductionCompaniesEntity)[] | null;
  production_countries?: (TmdbProductionCountriesEntity)[] | null;
  release_date: string;
  revenue: number;
  runtime: number;
  spoken_languages?: (TmdbSpokenLanguagesEntity)[] | null;
  status: string;
  tagline: string;
  title: string;
  video: boolean;
  vote_average: number;
  vote_count: number;
  translations?: TmdbTranslations;
  credits?: TmdbCredits;
}
export interface TmdbGenresEntity {
  id: number;
  name: string;
}
export interface TmdbProductionCompaniesEntity {
  id: number;
  logo_path?: string | null;
  name: string;
  origin_country: string;
}
export interface TmdbProductionCountriesEntity {
  iso_3166_1: string;
  name: string;
}
export interface TmdbSpokenLanguagesEntity {
  iso_639_1: string;
  name: string;
}
export interface TmdbTranslations {
  translations?: (TmdbTranslationEntity)[] | null;
}
export interface TmdbTranslationsEntity extends TmdbTranslations {
  id: number;
}
export interface TmdbTranslationEntity {
  iso_3166_1: string;
  iso_639_1: string;
  name: string;
  english_name: string;
  data: TmdbTranslationEntityData;
}
export interface TmdbTranslationEntityData {
  name?: string;
  title?: string;
  overview?: string;
  homepage?: string;
  biography?: string;
}
export interface TmdbCredits {
  cast?: (TmdbCastEntity)[] | null;
  crew?: (TmdbCrewEntity)[] | null;
}
export interface TmdbCreditsEntity extends TmdbCredits {
  id: number;
}
export interface TmdbCastEntity {
  cast_id: number;
  character: string;
  credit_id: string;
  gender: number;
  id: number;
  name: string;
  order: number;
  profile_path?: string | null;
}
export interface TmdbCrewEntity {
  credit_id: string;
  department: string;
  gender: number;
  id: number;
  job: string;
  name: string;
  profile_path?: string | null;
}
export interface TmdbPersonEntity {
  birthday: string;
  known_for_department: string;
  deathday?: null;
  id: number;
  name: string;
  movie_credits?: TmdbPersonMovieCredits;
  tv_credits?: TmdbPersonTvCredits;
  also_known_as?: (string)[] | null;
  gender: number;
  biography: string;
  popularity: number;
  place_of_birth: string;
  profile_path: string;
  adult: boolean;
  imdb_id: string;
  homepage?: string;
  translations?: TmdbTranslations;
}
export interface TmdbPersonMovieCredits {
  cast?: (TmdbPersonCastEntity)[] | null;
  crew?: (TmdbPersonPersonCrewEntity)[] | null;
}
export interface TmdbPersonCastEntity {
  release_date: string;
  adult: boolean;
  vote_average: number;
  vote_count: number;
  video: boolean;
  title: string;
  popularity: number;
  genre_ids?: (number | null)[] | null;
  original_language: string;
  character: string;
  original_title: string;
  poster_path?: string | null;
  id: number;
  backdrop_path?: string | null;
  overview: string;
  credit_id: string;
}
export interface TmdbPersonCrewEntity {
  id: number;
  department: string;
  original_language: string;
  original_title: string;
  job: string;
  overview: string;
  genre_ids?: (number)[] | null;
  video: boolean;
  credit_id: string;
  poster_path: string;
  popularity: number;
  backdrop_path: string;
  vote_count: number;
  title: string;
  adult: boolean;
  vote_average: number;
  release_date: string;
}
export interface TmdbPersonTvCredits {
  cast?: (TmdbPersonTvCastEntity)[] | null;
  crew?: (TmdbPersonTvCrewEntity)[] | null;
}
export interface TmdbPersonTvCastEntity {
  origin_country?: (string | null)[] | null;
  original_name: string;
  genre_ids?: (number | null)[] | null;
  vote_count: number;
  backdrop_path?: string | null;
  name: string;
  first_air_date: string;
  original_language: string;
  popularity: number;
  character: string;
  episode_count: number;
  id: number;
  credit_id: string;
  vote_average: number;
  overview: string;
  poster_path?: string | null;
}
export interface TmdbPersonTvCrewEntity {
  id: number;
  department: string;
  original_language: string;
  episode_count: number;
  job: string;
  overview: string;
  origin_country?: (string)[] | null;
  original_name: string;
  vote_count: number;
  name: string;
  popularity: number;
  credit_id: string;
  backdrop_path?: string | null;
  first_air_date: string;
  vote_average: number;
  genre_ids?: (number)[] | null;
  poster_path: string;
}

export interface TmdbTvEntity {
  backdrop_path: string;
  created_by?: (TmdbTvCreatedByEntity)[] | null;
  episode_run_time?: (number)[] | null;
  first_air_date: string;
  genres?: (TmdbGenresEntity)[] | null;
  homepage: string;
  id: number;
  in_production: boolean;
  languages?: (string)[] | null;
  last_air_date: string;
  last_episode_to_air: TmdbTvLastEpisodeToAir;
  name: string;
  next_episode_to_air: TmdbTvNextEpisodeToAir;
  networks?: (NetworksEntityOrProductionCompaniesEntity)[] | null;
  number_of_episodes: number;
  number_of_seasons: number;
  origin_country?: (string)[] | null;
  original_language: string;
  original_name: string;
  overview: string;
  popularity: number;
  poster_path: string;
  production_companies?:
    | (TmdbNetworksEntityOrProductionCompaniesEntity)[]
    | null;
  seasons?: (TmdbTvSeasonsEntity)[] | null;
  status: string;
  type: string;
  vote_average: number;
  vote_count: number;
  credits: TmdbTvCredits;
  translations: TmdbTranslations;
}
export interface TmdbTvCreatedByEntity {
  id: number;
  credit_id: string;
  name: string;
  gender: number;
  profile_path: string;
}
export interface TmdbTvLastEpisodeToAir {
  air_date: string;
  episode_number: number;
  id: number;
  name: string;
  overview: string;
  production_code: string;
  season_number: number;
  show_id: number;
  still_path: string;
  vote_average: number;
  vote_count: number;
}
export interface TmdbTvNextEpisodeToAir {
  air_date: string;
  episode_number: number;
  id: number;
  name: string;
  overview: string;
  production_code: string;
  season_number: number;
  show_id: number;
  still_path?: null;
  vote_average: number;
  vote_count: number;
}
export interface TmdbNetworksEntityOrProductionCompaniesEntity {
  name: string;
  id: number;
  logo_path: string;
  origin_country: string;
}
export interface TmdbTvSeasonsEntity {
  air_date: string;
  episode_count: number;
  id: number;
  name: string;
  overview: string;
  poster_path: string;
  season_number: number;
}
export interface TmdbTvCredits {
  cast?: (TmdbTvCastEntity)[] | null;
  crew?: (TmdbTvCrewEntity)[] | null;
}
export interface TmdbTvCastEntity {
  character: string;
  credit_id: string;
  id: number;
  name: string;
  gender: number;
  profile_path: string;
  order: number;
}
export interface TmdbTvCrewEntity {
  credit_id: string;
  department: string;
  id: number;
  name: string;
  gender: number;
  job: string;
  profile_path?: string | null;
}
