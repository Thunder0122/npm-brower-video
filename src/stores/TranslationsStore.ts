import { observable, computed, action } from "mobx";

import { TmdbTranslationEntity } from "../@types";

export interface TranslationsStoreInitialState {
  rawData: (TmdbTranslationEntity)[] | null;
}

const removeFalsy = <T>(obj: T): T => {
  let newObj: T = {} as T;
  Object.keys(obj).forEach(prop => {
    const key = prop as keyof typeof obj; // needed for typescript
    if (obj[key]) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
};

class TranslationsStore implements TranslationsStoreInitialState {
  @observable rawData: (TmdbTranslationEntity)[] | null = [];
  constructor(
    { rawData = [] }: { rawData: (TmdbTranslationEntity)[] | null } = {} as any
  ) {
    this.rawData = rawData;
  }
  @computed get availableLanguages() {
    return (this.rawData || []).map(translation => ({
      code: `${translation.iso_639_1}-${translation.iso_3166_1}`,
      label: `${translation.english_name}`
    }));
  }
  @action setTranslations(translations: (TmdbTranslationEntity)[]) {
    this.rawData = observable.array(translations);
  }
  @action resetTranslations() {
    this.rawData = observable.array([]);
  }
  /**
   * Retrieve missing translated parts of `data` using the tranlation infos.
   * 1) Injects data in `translationLanguageFullCode` if parameter passed and translation available in the language
   * 2) Injects data in `defaultLanguageFullCode` if data available in the language
   * 3) Injects data in `fallbackLanguageFullCode` if data available in the language
   */
  @action retrieveDataWithFallback<TData, TTranslationData>(
    data: TData,
    defaultLanguageFullCode: string,
    translationLanguageFullCode: string | undefined,
    fallbackLanguageFullCode = "en-US"
  ) {
    const match =
      this.rawData &&
      this.rawData.reduce(
        (acc, translation) => {
          [
            ["translationData", translationLanguageFullCode],
            ["defaultData", defaultLanguageFullCode],
            ["fallbackData", fallbackLanguageFullCode]
          ].forEach(([accumulatorKey, languageCode]) => {
            if (
              translation &&
              languageCode ===
                `${translation.iso_639_1}-${translation.iso_3166_1}`
            ) {
              (acc as any)[accumulatorKey as string] = removeFalsy(
                translation.data as TTranslationData
              );
            }
          });
          return acc;
        },
        { translationData: {}, defaultData: {}, fallbackData: {} } as {
          translationData: TTranslationData;
          defaultData: TTranslationData;
          fallbackData: TTranslationData;
        }
      );
    if (match) {
      // any missing data will be replaced by the default language
      const result = {
        ...data,
        ...match.fallbackData,
        ...match.defaultData,
        ...match.translationData
      };
      return result;
    }
    return { ...data };
  }
}

export default TranslationsStore;
