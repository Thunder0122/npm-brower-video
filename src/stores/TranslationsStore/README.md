# TranslationsStore

Mobx store that stores available translations for resources.

Sometimes, `english_name` only contains a [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) (two-letter codes, one per language) and not the full name.

This is why [`languageMapping.json`](./languageMapping.json) exists based on this [stackoverflow answer](https://stackoverflow.com/questions/3217492/list-of-language-codes-in-yaml-or-json/4900304#20623472) to provide the same language label for each code.
