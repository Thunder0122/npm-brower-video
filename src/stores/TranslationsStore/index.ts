import { observable, computed, action } from "mobx";

import languageMapping from "./languageMapping.json";
import { TmdbTranslationEntity } from "../../@types";

export interface TranslationsStoreInitialState {
  rawData: (TmdbTranslationEntity)[] | null;
}

const removeFalsy = <T>(obj: T): T => {
  let newObj: T = {} as T;
  Object.keys(obj).forEach(prop => {
    const key = prop as keyof typeof obj; // needed for typescript
    if (obj[key]) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
};

class TranslationsStore implements TranslationsStoreInitialState {
  @observable rawData: (TmdbTranslationEntity)[] | null = [];
  constructor(
    { rawData = [] }: { rawData: (TmdbTranslationEntity)[] | null } = {} as any
  ) {
    this.rawData = rawData;
  }
  @computed get availableLanguages() {
    return (this.rawData || [])
      .map(translation => ({
        code: `${translation.iso_639_1}-${translation.iso_3166_1}`,
        // sometimes, `english_name` only contains an iso-639-1 code
        label: `${
          translation.english_name.length > 2
            ? translation.english_name
            : (languageMapping as any)[translation.iso_639_1]
        } (${translation.iso_639_1}-${translation.iso_3166_1})`
      }))
      .sort((a, b) => (a.label > b.label ? 1 : -1));
  }
  @computed get availableLanguagesCodes() {
    return (this.rawData || [])
      .map(translation => `${translation.iso_639_1}-${translation.iso_3166_1}`)
      .sort();
  }
  @action setTranslations(translations: (TmdbTranslationEntity)[]) {
    this.rawData = observable.array(translations);
  }
  @action resetTranslations() {
    this.rawData = observable.array([]);
  }
  /**
   * Retrieve missing translated parts of `data` using the tranlation infos.
   * 1) Injects data in `translationLanguageFullCode` if parameter passed and translation available in the language
   * 2) Injects data in `defaultLanguageFullCode` if data available in the language
   * 3) Injects data in `fallbackLanguageFullCode` if data available in the language
   */
  @action retrieveDataWithFallback<TData, TTranslationData>(
    data: TData,
    defaultLanguageFullCode: string,
    translationLanguageFullCode: string | undefined,
    fallbackLanguageFullCode = "en-US"
  ) {
    const match =
      this.rawData &&
      this.rawData.reduce(
        (acc, translation) => {
          [
            ["translationData", translationLanguageFullCode],
            ["defaultData", defaultLanguageFullCode],
            ["fallbackData", fallbackLanguageFullCode]
          ].forEach(([accumulatorKey, languageCode]) => {
            if (
              translation &&
              languageCode ===
                `${translation.iso_639_1}-${translation.iso_3166_1}`
            ) {
              (acc as any)[accumulatorKey as string] = removeFalsy(
                translation.data as TTranslationData
              );
            }
          });
          return acc;
        },
        { translationData: {}, defaultData: {}, fallbackData: {} } as {
          translationData: TTranslationData;
          defaultData: TTranslationData;
          fallbackData: TTranslationData;
        }
      );
    if (match) {
      // any missing data will be replaced by the default language
      const result = {
        ...data,
        ...match.fallbackData,
        ...match.defaultData,
        ...match.translationData
      };
      return result;
    }
    return { ...data };
  }
}

export default TranslationsStore;
