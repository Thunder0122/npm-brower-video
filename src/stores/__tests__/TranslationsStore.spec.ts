import TranslationsStore from "../TranslationsStore";
import { TmdbMovieEntity } from "../../@types";
const movieFixture: TmdbMovieEntity = require("../../services/apis/__mocks__/tmdb/movie.fixtures.json");

describe("src/stores/TranslationsStore", () => {
  it("should store translations", () => {
    const store = new TranslationsStore();
    store.setTranslations(movieFixture.translations.translations);
    expect(store.rawData).toEqual(movieFixture.translations.translations);
  });
  it("should expose available languages", () => {
    const store = new TranslationsStore();
    store.setTranslations(movieFixture.translations.translations);
    expect(store.availableLanguages).toEqual(
      movieFixture.translations.translations.map(translation => ({
        code: `${translation.iso_639_1}-${translation.iso_3166_1}`,
        label: `${translation.english_name}`
      }))
    );
  });
  it("should epose available languages codes (sorted)", () => {
    const store = new TranslationsStore();
    store.setTranslations(movieFixture.translations.translations);
    expect(store.availableLanguagesCodes).toEqual(
      movieFixture.translations.translations
        .map(
          translation => `${translation.iso_639_1}-${translation.iso_3166_1}`
        )
        .sort()
    );
  });
  describe("retrieveDataWithFallback", () => {
    const defaultLanguageFullCode = "fr-FR";
    const translationLanguageFullCode = "es-ES";
    const fallbackLanguageFullCode = "en-US";
    it(`translationLanguageFullCode (${translationLanguageFullCode}) should take over`, () => {
      const store = new TranslationsStore();
      store.setTranslations(movieFixture.translations.translations);
      const translatedData = store.retrieveDataWithFallback(
        movieFixture,
        defaultLanguageFullCode,
        translationLanguageFullCode
      );
      expect({
        title: translatedData.title,
        overview: translatedData.overview
      }).toMatchSnapshot();
    });
    it(`partial fallback from translation language (${translationLanguageFullCode}) to default language (${defaultLanguageFullCode})`, () => {
      const store = new TranslationsStore();
      // remove the title from the es-ES translations
      store.setTranslations(
        movieFixture.translations.translations.map(translation => {
          if (
            `${translation.iso_639_1}-${translation.iso_3166_1}` !==
            translationLanguageFullCode
          ) {
            return translation;
          }
          return {
            ...translation,
            data: {
              ...translation.data,
              title: ""
            }
          };
        })
      );
      const translatedData = store.retrieveDataWithFallback(
        movieFixture,
        defaultLanguageFullCode,
        translationLanguageFullCode
      );
      expect({
        title: translatedData.title,
        overview: translatedData.overview
      }).toMatchSnapshot();
    });
    it(`full fallback from translation language (${translationLanguageFullCode}) to default language (${defaultLanguageFullCode}) - e.g. missing translations`, () => {
      const store = new TranslationsStore();
      // remove the es-ES translations
      store.setTranslations(
        movieFixture.translations.translations.filter(
          translation =>
            `${translation.iso_639_1}-${translation.iso_3166_1}` !==
            translationLanguageFullCode
        )
      );
      const translatedData = store.retrieveDataWithFallback(
        movieFixture,
        defaultLanguageFullCode,
        translationLanguageFullCode
      );
      expect({
        title: translatedData.title,
        overview: translatedData.overview
      }).toMatchSnapshot();
    });
    it(`partial fallback from default language (${defaultLanguageFullCode}) to fallback language (${fallbackLanguageFullCode})`, () => {
      const store = new TranslationsStore();
      // remove the es-ES translations and the title from the fr-FR
      store.setTranslations(
        movieFixture.translations.translations.map(translation => {
          if (
            `${translation.iso_639_1}-${translation.iso_3166_1}` !==
              translationLanguageFullCode &&
            `${translation.iso_639_1}-${translation.iso_3166_1}` !==
              defaultLanguageFullCode
          ) {
            return translation;
          }
          if (
            `${translation.iso_639_1}-${translation.iso_3166_1}` ===
            defaultLanguageFullCode
          ) {
            return {
              ...translation,
              data: {
                ...translation.data,
                title: ""
              }
            };
          }
        })
      );
      const translatedData = store.retrieveDataWithFallback(
        movieFixture,
        defaultLanguageFullCode,
        translationLanguageFullCode
      );
      expect({
        title: translatedData.title,
        overview: translatedData.overview
      }).toMatchSnapshot();
    });
    it(`full fallback from default language (${defaultLanguageFullCode}) to fallback language (${fallbackLanguageFullCode})`, () => {
      const store = new TranslationsStore();
      // remove the es-ES translations
      store.setTranslations(
        movieFixture.translations.translations.filter(
          translation =>
            `${translation.iso_639_1}-${translation.iso_3166_1}` !==
              translationLanguageFullCode &&
            `${translation.iso_639_1}-${translation.iso_3166_1}` !==
              defaultLanguageFullCode
        )
      );
      const translatedData = store.retrieveDataWithFallback(
        movieFixture,
        defaultLanguageFullCode,
        translationLanguageFullCode
      );
      expect({
        title: translatedData.title,
        overview: translatedData.overview
      }).toMatchSnapshot();
    });
  });
});
