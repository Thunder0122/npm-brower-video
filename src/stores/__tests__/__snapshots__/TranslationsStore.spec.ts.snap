// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/stores/TranslationsStore retrieveDataWithFallback full fallback from default language (fr-FR) to fallback language (en-US) 1`] = `
Object {
  "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
  "title": "Star Wars",
}
`;

exports[`src/stores/TranslationsStore retrieveDataWithFallback full fallback from translation language (es-ES) to default language (fr-FR) - e.g. missing translations 1`] = `
Object {
  "overview": "Il y a bien longtemps, dans une galaxie très lointaine... La guerre civile fait rage entre l'Empire galactique et l'Alliance rebelle. Capturée par les troupes de choc de l'Empereur menées par le sombre et impitoyable Dark Vador, la princesse Leia Organa dissimule les plans de l’Étoile Noire, une station spatiale invulnérable, à son droïde R2-D2 avec pour mission de les remettre au Jedi Obi-Wan Kenobi. Accompagné de son fidèle compagnon, le droïde de protocole C-3PO, R2-D2 s'échoue sur la planète Tatooine et termine sa quête chez le jeune Luke Skywalker. Rêvant de devenir pilote mais confiné aux travaux de la ferme, ce dernier se lance à la recherche de ce mystérieux Obi-Wan Kenobi, devenu ermite au cœur des montagnes désertiques de Tatooine...",
  "title": "La Guerre des étoiles",
}
`;

exports[`src/stores/TranslationsStore retrieveDataWithFallback partial fallback from default language (fr-FR) to fallback language (en-US) 1`] = `
Object {
  "overview": "Il y a bien longtemps, dans une galaxie très lointaine... La guerre civile fait rage entre l'Empire galactique et l'Alliance rebelle. Capturée par les troupes de choc de l'Empereur menées par le sombre et impitoyable Dark Vador, la princesse Leia Organa dissimule les plans de l’Étoile Noire, une station spatiale invulnérable, à son droïde R2-D2 avec pour mission de les remettre au Jedi Obi-Wan Kenobi. Accompagné de son fidèle compagnon, le droïde de protocole C-3PO, R2-D2 s'échoue sur la planète Tatooine et termine sa quête chez le jeune Luke Skywalker. Rêvant de devenir pilote mais confiné aux travaux de la ferme, ce dernier se lance à la recherche de ce mystérieux Obi-Wan Kenobi, devenu ermite au cœur des montagnes désertiques de Tatooine...",
  "title": "Star Wars",
}
`;

exports[`src/stores/TranslationsStore retrieveDataWithFallback partial fallback from translation language (es-ES) to default language (fr-FR) 1`] = `
Object {
  "overview": "La princesa Leia, líder del movimiento rebelde que desea reinstaurar la República en la galaxia en los tiempos ominosos del Imperio, es capturada por las malévolas Fuerzas Imperiales, capitaneadas por el implacable Darth Vader, el sirviente más fiel del emperador. El intrépido Luke Skywalker, ayudado por Han Solo, capitán de la nave espacial \\"El Halcón Milenario\\", y los androides, R2D2 y C3PO, serán los encargados de luchar contra el enemigo y rescatar a la princesa para volver a instaurar la justicia en el seno de la Galaxia.",
  "title": "La Guerre des étoiles",
}
`;

exports[`src/stores/TranslationsStore retrieveDataWithFallback translationLanguageFullCode (es-ES) should take over 1`] = `
Object {
  "overview": "La princesa Leia, líder del movimiento rebelde que desea reinstaurar la República en la galaxia en los tiempos ominosos del Imperio, es capturada por las malévolas Fuerzas Imperiales, capitaneadas por el implacable Darth Vader, el sirviente más fiel del emperador. El intrépido Luke Skywalker, ayudado por Han Solo, capitán de la nave espacial \\"El Halcón Milenario\\", y los androides, R2D2 y C3PO, serán los encargados de luchar contra el enemigo y rescatar a la princesa para volver a instaurar la justicia en el seno de la Galaxia.",
  "title": "La guerra de las galaxias. Episodio IV: Una nueva esperanza",
}
`;
